src/AWS.Deploy.Common/Recipes/Validation/OptionSettingItemValidators/InstanceTypeValidator.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Threading.Tasks;
+using AWS.Deploy.Common.Data;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Validates that a given EC2 instance is valid for the deployment region
+    /// </summary>
+    public class InstanceTypeValidator : IOptionSettingItemValidator
+    {
+        private readonly IAWSResourceQueryer _awsResourceQueryer;
+
+        public InstanceTypeValidator(IAWSResourceQueryer awsResourceQueryer)
+        {
+            _awsResourceQueryer = awsResourceQueryer;
+        }
+
+        public async Task<ValidationResult> Validate(object input, Recommendation recommendation)
+        {
+            var rawInstanceType = Convert.ToString(input);
+
+            // Instance type isn't required, so treat empty values as valid
```

Comment:
nit: I think the reason why this validator should return valid is not because of the instance type not being necessary in this case, but because we could add a Required validator if it was. I think it's better that we remove the comment.

Reply:
Dropped the comment.